lnd.go

```diff
@@ -259,6 +260,47 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 		return err
 	}
 
+	// Start leader election if we're running on etcd. Continuation will be
+	// blocked until this instance is elected as the current leader or
+	// shutting down.
+	elected := false
+	if cfg.Cluster.EnableLeaderElection &&
+		cfg.Cluster.LeaderElector == cluster.EtcdLeaderElector {
+		electionCtx, cancelElection := context.WithCancel(ctx)
+
+		go func() {
+			<-shutdownChan
+			cancelElection()
+		}()
+
+		leaderElector, err := cluster.MakeLeaderElector(
+			electionCtx, cfg.Cluster.LeaderElector,
+			cfg.Cluster.ID, cfg.Cluster.EtcdElectionPrefix,
+			cfg.DB.Etcd,
+		)
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if !elected {
+				return
+			}
+
+			if err := leaderElector.Resign(); err != nil {
+				ltndLog.Errorf("Leader elector failed to "+
+					"resign: %v", err)
+			}
+		}()
+
+		if err := leaderElector.Campaign(electionCtx); err != nil {
+			ltndLog.Errorf("Leadership campaign failed: %v", err)
+			return err
+		}
+
```

Comment:
Would be useful to add a log here after the instance becomes leader, and also before the `Campaign` method is called as well.

Reply:
Note that we do log these events just not here but in the `cluster` package. Can move if desired but wanted to keep all logs there to cover all branches properly.

Reply:
Note that there were logging for these events, but the logs were in the `cluster` package. I think they're probably better placed in lnd main, so moved them here.

Reply:
Ah I didn't realize, either way seems fine, slight preference in the `cluster` package to make it more general, non-blocking reallyimo as it's logging.